vmm/src/cpu.rs

```diff
@@ -919,12 +921,14 @@ impl CpuManager {
 
                                 #[cfg(feature = "kvm")]
                                 {
-                                    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(true);
-                                    if !matches!(vcpu.lock().unwrap().run(), Ok(VmExit::Ignore)) {
-                                        error!("Unexpected VM exit on \"immediate_exit\" run");
-                                        break;
+                                    if on_kvm {
+                                        vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(true);
+                                        if !matches!(vcpu.lock().unwrap().run(), Ok(VmExit::Ignore)) {
+                                            error!("Unexpected VM exit on \"immediate_exit\" run");
+                                            break;
+                                        }
+                                        vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(false);
                                     }
-                                    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(false);
                                 }
```

Comment:
Nit: it would be cleaner if you do:

```
#[cfg(feature = "kvm")]
if hypervisor_type == HypervisorType::Kvm {
    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(true);
    if !matches!(vcpu.lock().unwrap().run(), Ok(VmExit::Ignore)) {
        error!("Unexpected VM exit on \"immediate_exit\" run");
        break;
    }
    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(false);
}
```

Reply:
@russell-islam Please don't forget to address two nits here. Thanks.